reactive/nfs.py

```diff
@@ -50,27 +54,96 @@ def update_config():
         service_restart('nfs-kernel-server')
 
 
-@when('refresh_nfs_mounts')
+@when_any('refresh_nfs_mounts',
+          'config.changed.mount_options',
+          'config.changed.active_units')
 def read_nfs_mounts():
+    mount_interface = endpoint_from_name('nfs')
+    if mount_interface is None:
+        hookenv.log('No mount interface, bailing')
+        return
+
     hookenv.status_set('maintenance', 'Updating NFS mounts')
-    if service_running('nfs-kernel-server'):
+    service_is_running = service_running('nfs-kernel-server')
+    if service_is_running:
         try:
             command = ['exportfs', '-ra']
             hookenv.log('Executing {}'.format(command))
             check_output(command)
-            clear_flag('refresh_nfs_mounts')
         except CalledProcessError as e:
             hookenv.log(e)
             hookenv.log('Failed to reread nfs mounts. Will attempt again next update.')  # noqa
             return
+
+    config = hookenv.config()
+    storage_root = config.get('storage_root')
+    mount_options = config.get('mount_options')
+    active_ip = None
+    active_units = config.get('active_units', '')
+    need_service = True
+
+    if active_units:
+        # Work out which unit should be active so that we can publish its
+        # address on the mount relation.  We pick the first item in
+        # active_units that exists.
+        active_ip = None
+        peer_endpoint = endpoint_from_name('peer')
+        if peer_endpoint is not None:
+            peer_info = peer_endpoint.get_peer_info()
+            for active_unit in active_units.split(','):
```

Comment:
Might want to allow for whitespace in case the admin sets the config to something like `"nfs/0, nfs/1"`.  `re.split(r'\w*,\w*', active_units)` should do it.